src/main/resources/liquibase/202404051233-fix-bad-tally-state-and-event-data.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.20.xsd">
+  <changeSet id="202404051233-01" author="mstead">
+    <comment>
+      1. Copy the record_date value into the JSON data for records that currently have
+          data->'record_date' set to null.
+      2. Set the tally_state.latest_event_record_date to the end of March for any
+         record that has a null latest_event_record_date.
+    </comment>
+    <sql dbms="postgresql">
+        update events
+         set data = jsonb_set(data, '{record_date}', to_jsonb(record_date))
+         where data->'record_date' is null</sql>
+    <sql dbms="postgresql">
+        update tally_state
+        set latest_event_record_date = '2024-04-01T01:00Z'
+        where latest_event_record_date is null;
```

Comment:
does it make sense to add a non-null constraint on the "tally_state" table to prevent inserting invalid data in the future?

Reply:
Ya, it does make sense to add the non-null constraint. There's no case that I can think of where this should ever be null.

I've pushed the updated changeset. Please give the test steps another run through to make sure that it doesn't cause issues.